File: Tree/104.maximum-depth-of-binary-tree.java
/*
 * @lc app=leetcode id=104 lang=java
 *
 * [104] Maximum Depth of Binary Tree
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    int depth = 1;
    public int maxDepth(TreeNode root) {
        if(root == null) return 0;      
        if(root.left == null && root.right == null) return depth;
        return 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
    }
}
// @lc code=end

File: Tree/104.maximum-depth-of-binary-tree.py
#
# @lc app=leetcode id=104 lang=python3
#
# [104] Maximum Depth of Binary Tree
#

# @lc code=start
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        depth = 0
        if root == None:
            return depth
        return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))
# @lc code=end

